`defineFunction` docs (#6614)
aws-amplify/docs: src/pages/gen2/build-a-backend/functions/index.mdx

    entry?: string // optional path to the function code. Defaults to ./handler.ts
  */
});
```

Next, create `amplify/functions/my-demo-function/handler.ts`. This is where your function code will go.

```ts
// amplify/functions/my-demo-function/handler.ts

export const handler = async (event) => {
  // your function code goes here
  return 'You made a function!';
};
```

The handler file _must_ export an async function named "handler". This is the entry point to your function. For more information on writing functions, see https://docs.aws.amazon.com/lambda/latest/dg/nodejs-handler.html

Lastly, this function needs to be added to your backend.

```ts
// amplify/backend.ts
import { defineBackend } from '@aws-amplify/backend';
import { myDemoFunction } from './functions/my-demo-function/resource';

defineBackend({
  myDemoFunction
});
```

Now when you run `npx amplify sandbox` or deploy your app on Amplify, it will include your backend function. However, just defining a function doesn't do a whole lot. We need a way to invoke the function based on some event or request. Let's take a look at some examples to see how we can do this.

## Example - Create a Function trigger for Auth

> Auth has several events that can trigger functions to perform custom sign up, sign in and other tasks. Here we're going to configure a "preSignUp" trigger, but configuring any trigger is the same process.

Reviewer: ```suggestion
Auth has several events that can trigger functions to perform custom sign up, sign in and other tasks. In this example you will configure a "preSignUp" trigger.
```

> In our auth definition we can add the following.

Reviewer: ```suggestion
In our auth definition you can add the following:
```

we -> you

```ts
// amplify/auth/resource.ts

export const auth = defineAuth({
  loginWith: {
    email: true
  },
  triggers: {
    // configure a trigger to point to a function definition
    preSignUp: defineFunction({
      entry: './pre-sign-up-handler.ts'
    });
  }
});
```

Then create the function definition at `amplify/auth/pre-sign-up-handler.ts`.

```ts
// amplify/auth/pre-sign-up-handler.ts
import type { PreSignUpTriggerHandler } from 'aws-lambda';

export const handler: PreSignUpTriggerHandler = async (event) => {
  // your code here
};
```

<Callout info>

> Note: The `PreSignUpTriggerHandler` type comes from the [@types/aws-lambda](https://www.npmjs.com/package/@types/aws-lambda) NPM package. This package has types for all kinds of lambda event payloads that are useful for writing lambda functions.

Reviewer: ```suggestion
**Note:** The `PreSignUpTriggerHandler` type comes from the [@types/aws-lambda](https://www.npmjs.com/package/@types/aws-lambda) NPM package. This package contains types for different kinds of Lambda handlers, events, and responses.
```

bold "note" for consistency, minor tweak to the second sentence

</Callout>

Assuming that your auth definition is already added to your backend, you don't need to explicitly add your preSignUp function to the `defineBackend` properties. Since the preSignUp function is attached to the auth definition, Amplify is able to figure out that the function should be included in the deployment.

> Now when you deploy your app and a user signs up, this lambda will fire before the signup.

Reviewer: ```suggestion
Now when you deploy your app and a user signs up, this Lambda will fire before the signup.
```
